services/galley/src/Galley/App.hs

```diff
@@ -196,15 +201,17 @@ instance MonadHttp Galley where
 instance HasRequestId Galley where
   getRequestId = view reqId
 
-createEnv :: Metrics -> Opts -> IO Env
-createEnv m o = do
+createEnv :: (ClientState -> Opts -> IO (Maybe [TeamId])) -> Metrics -> Opts -> IO Env
```

Comment:
I don't think this works. The environment isn't created on every request, but only in `mkApp`, which itself is also called once in Galley.run.run